### PR TITLE
[dev-netstandard2.0] Add NetStandard2.0 and NetCoreApp12

### PR DESCRIPTION
**Pull request by ericstj**
Adds NetStandard2.0 TFM and makes it compatible with NetCoreApp12,
net461, and all Xamarin TFMs (which always get latest NetStandard).

/cc @rrelyea @emgarten @joelverhagen

### CODE REVIEW

**Comment by joelverhagen** [comment]
This looks good. Thanks for taking care of this. I'll run it through our CI to make sure all of the tests are passing.

**Review by emgarten — APPROVED** [review]

**Comment by joelverhagen** [comment]
Test failures:

```
  CompatibilityListProviderTests.CompatibilityListProvider_NetStandard15Supporting 	
 
Assert.Contains() Failure
Not found: DNX,Version=v4.6.2
In value:  String[] [".NETCoreApp,Version=v1.0", ".NETFramework,Version=v4.6.1", ".NETStandard,Version=v1.5", ".NETStandardApp,Version=v1.5", "DNX,Version=v4.6.1", ...]
   at NuGet.Frameworks.Test.CompatibilityListProviderTests.CompatibilityListProvider_NetStandard15Supporting() in /home/nugetagent/TeamCity/buildAgent/work/6146812267711dd9/test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityListProviderTests.cs:line 79

  CompatibilityListProviderTests.CompatibilityListProvider_NetStandard16Supporting 	
 
First failure:	dev-jver	#2903 	Changes (433) 	30 Nov 16 11:25
Assert.Contains() Failure
Not found: DNX,Version=v4.6.3
In value:  String[] [".NETCoreApp,Version=v1.0", ".NETFramework,Version=v4.6.1", ".NETStandard,Version=v1.6", ".NETStandardApp,Version=v1.6", "DNX,Version=v4.6.1", ...]
   at NuGet.Frameworks.Test.CompatibilityListProviderTests.CompatibilityListProvider_NetStandard16Supporting() in /home/nugetagent/TeamCity/buildAgent/work/6146812267711dd9/test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityListProviderTests.cs:line 121

  CompatibilityListProviderTests.CompatibilityListProvider_NetStandard17Supporting 	
 
Assert.Contains() Failure
Not found: DNX,Version=v4.6.3
In value:  String[] [".NETCoreApp,Version=v1.1", ".NETFramework,Version=v4.6.1", ".NETStandard,Version=v1.7", ".NETStandardApp,Version=v1.7", "DNX,Version=v4.6.1", ...]
   at NuGet.Frameworks.Test.CompatibilityListProviderTests.CompatibilityListProvider_NetStandard17Supporting() in /home/nugetagent/TeamCity/buildAgent/work/6146812267711dd9/test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityListProviderTests.cs:line 163
```

**Comment by ericstj** [comment]
Crud, thought I fixed those.  Let me clean it up.

**Comment by ericstj** [comment]
Should be fixed now.  I reran NuGet.Frameworks.Test and it passed.

**Comment by joelverhagen** [comment]
Cool, running the CI again.

**Comment by ericstj** [comment]
@joelverhagen did that pass?

**Comment by joelverhagen** [comment]
@ericstj, yup. Looks good! I will merge it to `dev-netstandard2.0`.

File: test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs [hunk 1]
```diff
@@ -84,13 +88,17 @@ public class CompatibilityTests
...
-        [InlineData("net461", "netstandard1.6", false)]
-        [InlineData("net461", "netstandard1.5", false)]
-        [InlineData("net461", "netstandard1.4", true)]
-        [InlineData("net461", "netstandard1.3", true)]
-        [InlineData("net461", "netstandard1.2", true)]
-        [InlineData("net461", "netstandard1.1", true)]
-        [InlineData("net461", "netstandard1.0", true)]
+
+        [InlineData("net462", "netstandard2.1", false)]
+        [InlineData("net462", "netstandard2.0", true)]
```
Comment (weshaggard) [inline test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs]: Should these be net461?

Comment (ericstj) [reply test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs]: Good catch, it should.

**Comment by ericstj** [comment]
ugh, need to rebase this, just a moment

**Comment by ericstj** [comment]
Ok, now should be minimal.

**Pull request merged**